packages/typegpu/src/name.ts

```diff
@@ -0,0 +1,53 @@
+import { $getNameForward } from './shared/symbols.ts';
+
+interface MetaData {
+  name: string | undefined;
+}
+
+declare global {
+  var __TYPEGPU_META__: WeakMap<object, MetaData>;
+}
+
+function isObject(value: unknown): value is object {
+  return !!value && (typeof value === 'object' || typeof value === 'function');
+}
+
+function isForwarded(value: object): value is { [$getNameForward]: object } {
+  if (!($getNameForward in value)) {
+    return false;
+  }
+  return isObject(value[$getNameForward]);
+}
+
+export function getName(definition: unknown): string | undefined {
+  if (!isObject(definition)) {
+    return undefined;
+  }
+  if (isForwarded(definition)) {
+    return getName(definition[$getNameForward]);
+  }
+  return globalThis.__TYPEGPU_META__?.get(definition)?.name;
+}
```

Comment:
It turns out that using a primitive as a key when retrieving a value from a WeakMap does not cause a runtime error, meaning we can omit the check for object-ness.

<img width="222" alt="image" src="https://github.com/user-attachments/assets/a55f5c65-584b-4467-aa62-1609cf37b1e8" />

```suggestion
export function getName(definition: unknown): string | undefined {
  if (isForwarded(definition)) {
    return getName(definition[$getNameForward]);
  }
  return (globalThis as unknown as GlobalWithMeta).__TYPEGPU_META__?.get(definition)?.name;
}
```